osi_logicallane.proto

```diff
@@ -0,0 +1,471 @@
+    // the items should be set as follows:
+    // * reference = URI to map, can remain empty if identical with definiton
+    //               in \c GroundTruth::map_reference
+    // * type = "net.asam.opendrive"
+    // * identifier[0] = id of t_road
+    // * identifier[1] = s of t_road_lanes_laneSection
+    // * identifier[2] = id of t_road_lanes_laneSection_left_lane,
+    //                         t_road_lanes_laneSection_right_lane
+    //
+    // \note For non-ASAM Standards, it is implementation-specific how
+    //       source_reference is resolved.
+    //
+    // \note The value has to be repeated, because one lane segment may be
+    //       derived from more than one origin segment. Multiple sources
+    //       may be added as reference as well, for example, a map and sensors.
+    //
+    repeated ExternalReference source_reference = 3;
+
+    // Reference to the physical lane this logical lane belongs to.
+    //
+    // This makes it possible to get detailed information on the physical lane
+    // properties, e.g. the visual colors of the boundaries, the road
+    // condition, etc..
+    //
+    // TODO Should this be a repeated field? Imagine the following scenario:
+    // ---------------------------------
+    //  l1                       l2
+    // ----------    l3        ---------
+    //  l4                       l5
+    // ---------------------------------
+    //
+    // In this case, we have five physical lanes: l1, l2, l4 and l5 where a
+    // lane marking is present. And l3 is one lane covering the whole road
+    // (because no road marking is present).
+    //
+    // Now there are two basic possibilities to define the logical lanes:
+    // - There could be two logical lanes: one encompassing l4, part of l3 and
+    //   l5; another encompassing l2, part of l3 and l1. In this case, one
+    //   logical lane can map to a number of physical lanes, and the mapping
+    //   should include s positions, so it is clear which part of the logical
+    //   lane maps to which physical lane.
+    // - There could be 6 logical lanes: l1, l2, l4, l5 and two lanes for each
+    //   halve of l3. Then each logical lane would map to a single physical
+    //   lane, but logical lanes would have to be split each time a lane
+    //   marking appears or disappears.
+    //
+    optional Identifier physical_lane_id = 4;
+
+    // The reference line for this logical lane
+    //
+    // The reference line is used as a coordinate system on this lane.
+    //
+    // The reference line should roughly have the same shape as the lane, so
+    // that S coordinates continually increase/decrease along the lane. It is
+    // not required that the reference line has the same direction as the
+    // driving direction of the lane.
+    //
+    // Neighbouring lanes are strongly encouraged to use the same
+    // reference_line, so that vehicles that are next to each other on
+    // neighbouring lanes have comparable S positions.
```

Comment:
neighbour relation vs overlapping relation: how to define neighbour relations when lanes may overlap? When is a lane the right/left neighbour?

Reply:
Clarified the text. Probably needs an image to explain.

Reply:
Added an image showing both overlapping and neighbour lanes.